src/event/EventManager.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020-2023 Delft University of Technology and SynTest contributors
+ *
+ * This file is part of SynTest Framework - SynTest Core.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Encoding } from "..";
+import { PluginInterface } from "./PluginInterface";
+import { ProgramState } from "./ProgramState";
+
+export class EventManager<T extends Encoding> {
+  private _state: ProgramState<T>;
+  private listeners: PluginInterface<T>[] = [];
+
+  constructor(state: ProgramState<T>) {
```

Comment:
We would initialize the program state both here and in the launcher.  This would create two separate instances

Reply:
No, since we pass by reference this will be the same object.
I only have it in the launcher such that I do not have to call this.eventmanager.programstate.targetpool, but instead just use this.programstate.targetpool.

I guess it depends on how clean the code should be.

Reply:
Fixed.